modules/data.atmosphere/R/read.register.R

```diff
@@ -35,10 +35,14 @@ read.register <- function(register.xml, con) {
     } else if ((!is.null(register$format$id) & is.null(register$format$name)) 
                | 
                (!is.null(register$format$id) & is.null(register$format$mimetype))) {
-      register$format$name <- PEcAn.DB::db.query(
-        paste("SELECT name from formats where id = ", register$format$id), con)[[1]]
-      register$format$mimetype <- PEcAn.DB::db.query(
-        paste("SELECT mime_type from formats where id = ", register$format$id), con)[[1]]
+      # get name and mime_type from formats table in a singular dataframe
+      format.info <- PEcAn.DB::db.query(
+        paste("SELECT name, mime_type FROM formats WHERE id = ", register$format$id), con
+      )
+
+      # Assign the values from the data frame to the respective columns in register$format
+      register$format$name <- format.info$name
+      register$format$mimetype <- format.info$mime_type
```

Comment:
Looks like this could be done in one line, register$format = format.info, if you do the name change in the query, mime_type as mimetype.